### PR TITLE
tapsetup: add support for single, unbridged TUN interface 

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Sadly, TUN interfaces don't seem to be bridgeable (at least in Linux, every time I try to add a TUN to a bridge I get an invalid argument error). However, to go forward with https://github.com/RIOT-OS/RIOT/issues/14618, we should add a TUN support to the `dist/tools/tapsetup/tapsetup` script to streamline and unify the process of TAP/TUN interface handling.

### Testing procedure
Only tested on Linux so far, but basically used the several commands of `tapsetup` (`-d`, `-c`, `-l`, `-f`, ...) with the `--tun` argument and checked if everything works.

### Issues/PRs references
Addresses parts of https://github.com/RIOT-OS/RIOT/issues/14618
Addresses things discussed in https://github.com/RIOT-OS/RIOT/issues/14689
Follow-up on #14928

### CODE REVIEW

**Comment by benpicco** [comment]
Works as advertised (I only have Linux at hand too).
One small blemish: `tapbr0` gets deleted twice

```
% sudo dist/tools/tapsetup/tapsetup -d    
deleting tapbr0
deleting tap0
deleting tap1
deleting tapbr0
```

Do you plan on adding the option to add a single TAP interface (without a bridge) too, such as `ethos` or the [`gnrc_networking`](https://github.com/RIOT-OS/RIOT/tree/master/examples/gnrc_networking#connecting-riot-native-and-the-linux-host) example use?

**Comment by miri64** [comment]
> One small blemish: `tapbr0` gets deleted twice

Curious, will check why.

> Do you plan on adding the option to add a single TAP interface (without a bridge) too, such as `ethos` or the [`gnrc_networking`](https://github.com/RIOT-OS/RIOT/tree/master/examples/gnrc_networking#connecting-riot-native-and-the-linux-host) example use?

Actually no, I think connection between host and node should be as unified as possible, so my actual plan is to use `tapsetup` (including the set-up bridge) for the unified `setup_network.sh` to use for `ethos`

File: dist/tools/tapsetup/tapsetup [hunk 1]
```diff
@@ -213,73 +234,103 @@ del_ipv6_routes() {
...
+        FreeBSD|OSX)
+            ifconfig "${IFACE}" destroy ;;
+        Linux)
+            ip link delete "${IFACE}" ;;
+        *) ;;
+    esac
+}
+
 delete_bridge() {
     echo "deleting ${BRNAME}"
```
Comment (benpicco) [inline dist/tools/tapsetup/tapsetup:250]:
```suggestion
```

`delete_iface` already does the printing - you can drop this.

Comment (miri64) [reply dist/tools/tapsetup/tapsetup:250]: Done

**Review by benpicco — APPROVED** [review]

**Comment by benpicco** [comment]
Now I wonder if we should rename it to `tuntapsetup` to reflect it's newly gained capabilities.

**Comment by miri64** [comment]
Naaah, the main feature is still "create bridged TAP interfaces". Also try google for stuff like "add tun interface to bridge". Even Big G tries to correct that to "tap" ^^.

**Comment by benpicco** [comment]
But now we have `tapsetup` to set up a TUN interface.

**Comment by miri64** [comment]
Only if the _optional_ `--tun` argument is provided. IMHO, it would be confusing, mainly for legacy reasons, to rename that script.... I'm still struggling with the change that "sudo" is required in the call. Not to mention all the doc here, in other repos, and in introductory courses, that needs to be changed for that.

**Pull request merged**